docs/developer/eventing/sinks/creating-event-sinks/writing-event-sinks-with-golang/README.md

```diff
@@ -0,0 +1,330 @@
+func processEvent(e cloudevents.Event) (interface{} /*result*/, error) {
+	tBegin := time.Now()
+
+	// simulate processing time
+	randomDelay()
+
+	res := &exampleResult{
+		Message:        "event processed successfully",
+		ProcessingTime: time.Since(tBegin).Milliseconds(),
+	}
+
+	return res, nil
+}
+
+// exampleResult represents a fictional structured result of some event
+// processing.
+type exampleResult struct {
+	Message        string `json:"message"`
+	ProcessingTime int64  `json:"processing_time_ms"`
+}
+
+// newAckEvent returns a CloudEvent that acknowledges the processing of an
+// event.
+func newAckEvent(e cloudevents.Event, code ACKCode, detail interface{}) *cloudevents.Event {
+	resp := cloudevents.NewEvent()
+	resp.SetType(eventTypeAck)
+	resp.SetSource(eventSrcName)
+	resp.SetExtension(ceExtProcessedType, e.Type())
+
+	data := &ACKResponse{
+		Code:   code,
+		Detail: detail,
+	}
+
+	if err := resp.SetData(cloudevents.ApplicationJSON, data); err != nil {
+		log.Panicf("Error serializing CloudEvent data: %s", err)
+	}
+
+	return &resp
+}
+
+// randomDelay sleeps for a random amount of time.
+func randomDelay() {
+	const maxDelay = 100 * time.Millisecond
+
+	randomDelay := time.Duration(rand.Int63n(int64(maxDelay)))
+	time.Sleep(randomDelay)
+}
+
+```
+
+Create the `go.mod` file and run `tidy`:
+```
+go mod init
+go mod tidy
+```
+
+Create a new file `Dockerfile` in the same folder:
+```dockerfile
+FROM golang:1.15-buster AS builder
+ENV CGO_ENABLED 0
+WORKDIR /project
+COPY . ./
+RUN go build
+FROM gcr.io/distroless/static:nonroot
+COPY --from=builder /project/demosink /
+ENTRYPOINT ["/demosink"]
+```
+
+Run the application locally to debug:
+
+```sh
+go run .
+```
+
+Send it an event and verify the expected results:
+```sh
+curl -v "http://localhost:8080" \
+       -X POST \
+       -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+       -H "Ce-Specversion: 1.0" \
+       -H "Ce-Type: io.demo.email.send" \
+       -H "Ce-Source: dev.knative.samples/demo" \
+       -H "Content-Type: application/json" \
+       -d '{"fromName":"richard","toName":"bob","message":"hello"}'
+* upload completely sent off: 55 out of 55 bytes
+< HTTP/1.1 200 OK
+< Ce-Id: 70eb24d9-3678-4c08-9332-315cbae7fe1e
+< Ce-Processedid: 536808d3-88be-4077-9d7a-a3f162705f79
+< Ce-Processedsource: dev.knative.samples/demo
+< Ce-Processedtype: io.demo.email.send
+< Ce-Source: io.demo.targets.go-sample
+< Ce-Specversion: 1.0
+< Ce-Time: 2021-09-08T19:40:37.1619Z
+< Ce-Type: com.example.target.ack
+< Content-Length: 98
+< Content-Type: application/json
+< Date: Wed, 08 Sep 2021 19:40:37 GMT
+<
+* Connection #0 to host localhost left intact
+{"code":0,"detail":{"message":"Hello richard! Thank you for the message!","processing_time_ms":0}}* Closing connection 0
+```
+
+
+When you are ready to deploy, build and publish the container image:
+
+Google:
+```sh
+gcloud builds submit --tag gcr.io/<project-name>/<image-name> .
```

Comment:
All of our documentation should be vendor neutral, so if this sample requires Google platform specific tools etc please modify it so that there is a generic version instead, e.g. using `kubectl` commands.

Reply:
Fixed by removing the `gcloud` reference: https://github.com/knative/docs/pull/4208/commits/891e4c9ba7c5f72755deebdae93b81a2c7866c73 
Thank you!